network/src/transport.rs

```diff
@@ -132,56 +143,56 @@ pub struct Connection<TSocket> {
     pub metadata: ConnectionMetadata,
 }
 
-fn identity_key_to_peer_id(
-    trusted_peers: &RwLock<HashMap<PeerId, NetworkPublicKeys>>,
-    remote_static_key: &[u8],
-) -> Option<PeerId> {
-    for (peer_id, public_keys) in trusted_peers.read().unwrap().iter() {
-        if public_keys.identity_public_key.to_bytes() == remote_static_key {
-            return Some(*peer_id);
-        }
+/// Convert a remote pubkey into a network `PeerId`.
+fn identity_pubkey_to_peer_id(
+    maybe_trusted_peers: Option<&Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>>,
```

Comment:
why call it `maybe_*` when the `Option` already indicates this?

Reply:
it does, but I think it's confusing when in some places `trusted_peers: Arc<..>` and in others `trusted_peers: Option<Arc<..>>`

Reply:
actually as an API name I do find it kinda repetitive but I could see it being called maybe within a function body but your call

Reply:
agree here ^ it would be useful as a variable name, but I don't think it's idiomatic to have it as an argument name in the function signature